src/BlockTypes/ProductButton.php

```diff
@@ -144,53 +144,50 @@ protected function render( $attributes, $content, $block ) {
 				$args['attributes']['aria-label'] = wp_strip_all_tags( $args['attributes']['aria-label'] );
 			}
 
-			$div_directives    = 'data-wc-context=\'' . wp_json_encode( $context ) . '\'';
+			$div_directives    = 'data-wc-context=\'' . wp_json_encode( $context, JSON_NUMERIC_CHECK ) . '\'';
 			$button_directives = '
 				data-wc-on--click="actions.woocommerce.addToCart"
 				data-wc-class--loading="context.woocommerce.isLoading"
 				data-wc-class--added="selectors.woocommerce.isThereMoreThanOneItem"
-				data-wc-text="selectors.woocommerce.addToCartText"
 			';
 
-			/**
-			 * Filters the add to cart button class.
-			 *
-			 * @since 8.7.0
-			 *
-			 * @param string $class The class.
-			 */
-			return apply_filters(
-				'woocommerce_loop_add_to_cart_link',
-				strtr(
-					'<div class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
-						{div_directives}
-					>
-					<{html_element}
-						href="{add_to_cart_url}"
-						class="{button_classes}"
-						style="{button_styles}"
-						{attributes}
-						{button_directives}
-					>
-						{add_to_cart_text}
-					</{html_element}>
-					{view_cart_anchor}
-				</div>',
-					array(
-						'{classes}'           => esc_attr( $text_align_styles_and_classes['class'] ?? '' ),
-						'{custom_classes}'    => esc_attr( $classname . ' ' . $custom_width_classes ),
-						'{html_element}'      => $html_element,
-						'{add_to_cart_url}'   => esc_url( $product->add_to_cart_url() ),
-						'{button_classes}'    => isset( $args['class'] ) ? esc_attr( $args['class'] . ' ' . $is_added_class ) : $is_added_class,
-						'{button_styles}'     => esc_attr( $styles_and_classes['styles'] ),
-						'{attributes}'        => isset( $args['attributes'] ) ? wc_implode_html_attributes( $args['attributes'] ) : '',
-						'{add_to_cart_text}'  => esc_html( $initial_product_text ),
-						'{div_directives}'    => $is_ajax_button ? $div_directives : '',
-						'{button_directives}' => $is_ajax_button ? $button_directives : '',
-						'{view_cart_anchor}'  => $is_ajax_button ? $this->get_view_cart_html() : '',
-					)
-				),
+			$span_button_directives = 'data-wc-text="selectors.woocommerce.addToCartText"';
+
+			if ( WC()->cart->is_empty() ) {
```

Comment:
This is causing me an error when saving posts. It looks like `WC()->cart` is null, so the `is_empty()` method doesn't exist. Can you reproduce as well?

![imatge](https://github.com/woocommerce/woocommerce-blocks/assets/3616980/bc9e1007-121e-468d-aee1-e06a0d491d2c)